src/python/pants/backend/python/dependency_inference/default_module_mapping.py

```diff
@@ -186,41 +201,58 @@ def two_groups_hyphens_two_replacements_with_suffix(
     "paho-mqtt": ("paho",),
     "phonenumberslite": ("phonenumbers",),
     "pillow": ("PIL",),
+    "pillow-heif": ("pillow_heif",),
     "pip-tools": ("piptools",),
     "progressbar2": ("progressbar",),
     "protobuf": ("google.protobuf",),
     "psycopg2-binary": ("psycopg2",),
     "py-healthcheck": ("healthcheck",),
     "pycrypto": ("Crypto",),
+    "pycryptodome": ("Crypto",),
+    "pyerfa": ("erfa",),
     "pygithub": ("github",),
     "pygobject": ("gi",),
     "pyhamcrest": ("hamcrest",),
     "pyjwt": ("jwt",),
     "pykube-ng": ("pykube",),
     "pymongo": ("bson", "gridfs", "pymongo"),
     "pymupdf": ("fitz",),
+    "pynacl": ("nacl",),
     "pyopenssl": ("OpenSSL",),
     "pypdf2": ("PyPDF2",),
     "pypi-kenlm": ("kenlm",),
+    "pyshp": ("shapefile",),
     "pysocks": ("socks",),
     "pytest": ("pytest", "_pytest"),
+    "pytest-benchmark": ("pytest_benchmark",),
     "pytest-runner": ("ptr",),
+    "python-sat": ("pysat",),
+    "python-dateutil": ("dateutil",),
     "python-json-logger": ("pythonjsonlogger",),
     "python-levenshtein": ("Levenshtein",),
     "python-lsp-jsonrpc": ("pylsp_jsonrpc",),
+    "python-magic": ("magic",),
+    "python-solvespace": ("python_solvespace",),
     "pywinrm": ("winrm",),
+    "pywavelets": ("pywt",),
     "pyyaml": ("yaml",),
     "randomwords": ("random_words",),
+    "robotraconteur": ("RobotRaconteur",),
+    "ruamel-yaml": ("ruamel",),
```

Comment:
Oh. This one is sadly quite complicated :cold_sweat:.

First, I believe the canonical name for this package is `ruamel.yaml`. I'm not sure how many variants of that (like `ruamel-yaml`) pypi will happily accept as aliases for it. Both of these redirect to `ruamel.yaml`:
- https://pypi.org/project/ruamel-yaml/
- https://pypi.org/project/ruamel_yaml/

`ruamel` is a namespace package shared by a variety of `ruamel.*` packages maintained by Anthon van der Neut. `ruamel.yaml` is probably the most common package under the `ruamel` namespace package (source is available [on sourceforge](https://sourceforge.net/p/ruamel-yaml/code/). I think decent dep inferrence will need to account for more than just `ruamel` to tell which 3rd party package some 1st party code depends on.

To make matters worse (or at least, to add to the complexity), there are even deeper subpackages:
- https://pypi.org/project/ruamel.yaml/
- https://pypi.org/project/ruamel.yaml.clib/
- https://pypi.org/project/ruamel.yaml.cmd/
- https://pypi.org/project/ruamel.yaml.jinja2/

@huonw do you know if we have special handling for namespace packages? Maybe this should be a pattern, but I think dep inference only looks at the first package in an import.

For kicks and giggles, feel free to peruse the custom `NameSpacePackager` and related code in `ruamel.yaml`'s setup.py (which also gets metadata from __init__.py):
- https://sourceforge.net/p/ruamel-yaml/code/ci/default/tree/setup.py
- https://sourceforge.net/p/ruamel-yaml/code/ci/default/tree/__init__.py

Reply:
Hmm. I see the opentelemetry examples above. Maybe we need something like that:
```
    "opentelemetry-exporter-otlp": ("opentelemetry.exporter.otlp",),
    "opentelemetry-exporter-otlp-proto-grpc": ("opentelemetry.exporter.otlp.proto.grpc",),
```
So
```
    "ruamel.yaml": ("ruamel.yaml",),
    "ruamel.yaml.clib": ("ruamel.yaml.clib",),
    "ruamel.yaml.jinja2": ("ruamel.yaml.jinja2",),
```

Reply:
hmm not sure but 
```
"ruamel.yaml": ("ruamel.yaml",),  
    "ruamel.yaml.clib": ("ruamel.yaml.clib",),  
    "ruamel.yaml.jinja2": ("ruamel.yaml.jinja2",),  
```
seems identical so i removed them for now..

Reply:
> @huonw do you know if we have special handling for namespace packages? Maybe this should be a pattern, but I think dep inference only looks at the first package in an import.

I don't know how that works.